src/ucp/tag/offload/eager.c

```diff
@@ -326,6 +327,8 @@ ucp_proto_eager_sync_tag_offload_zcopy_send_completion(uct_completion_t *self)
                                           send.state.uct_comp);
 
     ucp_proto_request_zcopy_clean(req, UCS_BIT(UCP_DATATYPE_CONTIG));
+    ucp_datatype_iter_cleanup(&req->send.state.dt_iter,
+                              UCS_BIT(UCP_DATATYPE_CONTIG));
```

Comment:
is it needed?

Reply:
Basically, this is a common approach for all protocols to cleanup iterator on completion, the fact of `UCP_DATATYPE_CONTIG` iterator cleanup is noop, it's implementation details which can be changed, so I think it should be here.

Reply:
ok, missed that it's not reset